src/server/server.js

```diff
@@ -124,6 +126,28 @@ function processPluginHtml(html, pluginId) {
     });
 }
 
+function proxyXHR(request, response) {
+    var requestURL = url.parse(unescape(request.query.rurl));
+    
+    request.headers.host = requestURL.host;
+    // fixes encoding issue
+    delete request.headers["accept-encoding"];
+
+    var options = {
+        host: requestURL.host,
+        path: requestURL.path,
+        port: requestURL.port,
+        method: request.method,
+        headers: request.headers
+    };
+    
+    var proxyCallback = function(proxyReponse) {
+        proxyReponse.pipe(response);
+    };
+    
+    http.request(options, proxyCallback).end();
+}
+
```

Comment:
This features, can be done as middlewares. We keep the server as minimum as possible, as a core, and everything we can add as an extra/optional, can be a middleware.
We can implement proxyXHR as part of another module (express middleware), and then use it only if the feature is enabled. As it is right now, we are assuming in the server-side that we need to resolve CORS, when it is not necessary true.